File: baekjoon/bruteforce/_1339/gyeoul_1339.java
package baekjoon.bruteforce._1339;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.Collections;

public class gyeoul_1339 {
    public static void main(String[] args) throws Exception {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        ArrayList<Integer> list = new ArrayList<>();
        for (int i = 0; i < 26; i++) {
            list.add(0);
        }

        int num = Integer.parseInt(bf.readLine());
        for (int i = 0; i < num; i++) {
            char[] chars = bf.readLine().toCharArray();
            for (int j = 1; j <= chars.length; j++) {
                int multiplier = 1;
                for (int k = 1; k < j; k++) {
                    multiplier *= 10;
                }
                int index = chars[chars.length - j] - 'A';
                list.set(index, list.get(index) + multiplier);
            }
        }

        Collections.sort(list);
        Collections.reverse(list);

        int answer = 0;
        for (int i = 1; i <= 10; i++) {
            answer += list.get(i - 1) * (10 - i);
        }
        bw.write(""+answer);
        bw.flush();
        bw.close();
        bf.close();
    }
}

File: baekjoon/bruteforce/_1339/gyeoul_1339.kt
package baekjoon.bruteforce._1339

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

val bf = BufferedReader(InputStreamReader(System.`in`))
val bw = BufferedWriter(OutputStreamWriter(System.out))

fun main() {
    val rawNum = bf.readLine().toInt()
    val charMap = HashMap<Char, Int>()
    for (c in 'A'..'Z') {
        charMap[c] = 0
    }
    for (i in 0 until rawNum) {
        val it = bf.readLine().split("")
        val temp = it.subList(1, it.size - 1)
        for (j in temp.indices) {
            var value = 1
            for (k in 0 until temp.size - j - 1) value *= 10
            charMap[temp[j].toCharArray()[0]] = charMap[temp[j].toCharArray()[0]]!! + value
        }
    }
    val charList = charMap.toList().sortedByDescending { it.second }

//    bw.write("$charMap\n")
//    bw.write(charList.toString()+"\n")
    var sum = 0
    for (i in 0..9) {
        sum += (charList[i].second * (9 - i))
    }
    bw.write(sum.toString() + "\n")

//    bw.write(charArr.contentToString() +"\n")
//    bw.write(charMap.toString()+"\n")

    bw.flush()
    bw.close()
    bf.close()
}
